kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -181,10 +191,80 @@ impl GotocCodegenBackend {
+    /// find or create the `AssignsContract` that needs to be enforced and attach it to the symbol
+    /// for which it needs to be enforced.
+    ///
+    /// 1. Gets the `#[kanitool::inner_check = "..."]` target, then resolves exactly one instance
+    ///    of it. Panics if there are more or less than one instance.
+    /// 2. Expects that a `#[kanitool::modifies(...)]` is placed on the `inner_check` function,
+    ///    turns it into a CBMC contract and attaches it to the symbol for the previously resolved
+    ///    instance.
+    /// 3. Returns the mangled name of the symbol it attached the contract to.
+    /// 4. Resolves the `#[kanitool::checked_with = "..."]` target from `function_under_contract`
+    ///    which has `static mut REENTRY : bool` declared inside.
+    /// 5. Returns the full path to this constant that `--nondet-static-exclude` expects which is
+    ///    comprised of the file path that `checked_with` is located in, the name of the
+    ///    `checked_with` function and the name of the constant (`REENTRY`).
+    fn handle_check_contract(
+        &mut self,
+        function_under_contract: DefId,
+        items: &[MonoItem<'tcx>],
+    ) -> AssignsContract {
+        let tcx = self.tcx;
+        let function_under_contract_attrs = KaniAttributes::for_item(tcx, function_under_contract);
+        let wrapped_fn = function_under_contract_attrs.inner_check().unwrap().unwrap();
+
+        let mut instance_under_contract = items.iter().copied().filter_map(|i| match i {
+            MonoItem::Fn(instance @ Instance { def: InstanceDef::Item(did), .. }) => {
+                (wrapped_fn == did).then_some(instance)
+            }
+            _ => None,
+        });
+        let instance_of_check = instance_under_contract.next().unwrap();
+        assert!(
+            instance_under_contract.next().is_none(),
+            "Only one instance of a checked function may be in scope"
+        );
+        let attrs_of_wrapped_fn = KaniAttributes::for_item(tcx, wrapped_fn);
+        let assigns_contract = attrs_of_wrapped_fn.modifies_contract().unwrap_or_else(|| {
+            debug!(?instance_of_check, "had no assigns contract specified");
+            vec![]
+        });
+        self.attach_contract(instance_of_check, assigns_contract);
+
+        let wrapper_name = tcx.symbol_name(instance_of_check).to_string();
+
+        let recursion_wrapper_id =
+            function_under_contract_attrs.checked_with_id().unwrap().unwrap();
+        let span_of_recursion_wrapper = tcx.def_span(recursion_wrapper_id);
+        let location_of_recursion_wrapper = self.codegen_span(&span_of_recursion_wrapper);
+
+        let full_name = format!(
+            "{}:{}::REENTRY",
+            location_of_recursion_wrapper
+                .filename()
+                .expect("recursion location wrapper should have a file name"),
+            tcx.item_name(recursion_wrapper_id),
+        );
```

Comment:
In which case we would ended up without a file name at this point?

Reply:
Are you asking in which case this can fail? I'm honestly not sure. I assume only Kani or CBMC builtin symbols might not have a filename but those you couldn't attach a contract to anyway.